ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -347,13 +358,33 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
           close(key);
         }
       }
+      completeSslHandshakes();
```

Comment:
this call should really happen before line 321, shouldn't it? why bother checking those connections that were just added to the list?